File: app/src/main/java/ru/gsench/githubusers/domain/usecase/UserListUseCase.java
package ru.gsench.githubusers.domain.usecase;

import ru.gsench.githubusers.domain.interactor.UserModel;
import ru.gsench.githubusers.presentation.presenter.UserListPresenter;

/**
 * Created by grish on 06.05.2017.
 */

public interface UserListUseCase {
    public void subscribe(UserListPresenter presenter);
    public void updateList(int limit, int offset);
    public void openUser(UserModel user);
    public void pushFavorite(UserModel user);
}

File: app/src/main/java/ru/gsench/githubusers/domain/usecase/UserListUseCase.kt
package ru.gsench.githubusers.domain.usecase

import ru.gsench.githubusers.domain.interactor.UserModel
import ru.gsench.githubusers.presentation.presenter.UserListPresenter

/**
 * Created by grish on 06.05.2017.
 */

interface UserListUseCase {
    fun subscribe(presenter: UserListPresenter)
    fun updateList(limit: Int, offset: Int)
    fun openUser(user: UserModel)
    fun pushFavorite(user: UserModel)
}
